velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -207,6 +207,53 @@ struct UnixTimestampParseWithFormatFunction
   bool invalidFormat_{false};
 };
 
+/// Parse unix time in seconds to a string in given time format.
+template <typename T>
+struct FromUnixtimeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  const date::time_zone* sessionTimeZone_ = nullptr;
+  std::shared_ptr<DateTimeFormatter> mysqlDateTime_;
+  bool isConstantTimeFormat = false;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<int64_t>* /*unixtime*/,
+      const arg_type<Varchar>* timeFormat) {
+    sessionTimeZone_ = getTimeZoneFromConfig(config);
+    if (timeFormat != nullptr) {
+      isConstantTimeFormat = true;
+      mysqlDateTime_ = buildJodaDateTimeFormatter(
+          std::string_view(timeFormat->data(), timeFormat->size()));
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<int64_t> second,
+      const arg_type<Varchar> timeFormat) {
+    if (!isConstantTimeFormat) {
+      mysqlDateTime_ = buildJodaDateTimeFormatter(
+          std::string_view(timeFormat.data(), timeFormat.size()));
+    }
+    Timestamp timestamp = Timestamp::fromSecond(second);
```

Comment:
Seems the extra method `fromSecond` is not frequently used, and does not bring more convenience than `Timestamp(second, 0)`.

Reply:
Just removed. Thanks!